mldatafind/io.py

```diff
@@ -188,39 +196,81 @@ def read_timeseries(
     return data, times
 
 
+def _fetch_open_data(
+    ifos: List[str], start: float, end: float, verbose: bool, **kwargs
+) -> TimeSeriesDict:
+    ts_dict = TimeSeriesDict()
+    for ifo in ifos:
+        ts_dict[ifo] = TimeSeries.fetch_open_data(
+            ifo, start, end, verbose=verbose, **kwargs
+        )
+    return ts_dict
+
+
 def fetch_timeseries(
     channels: List[str],
-    t0: float,
-    tf: float,
-    nproc: int = 1,
+    start: float,
+    end: float,
     array_like: bool = False,
+    verbose: bool = False,
+    **kwargs,
 ) -> Union[TimeSeriesDict, Tuple[np.ndarray, np.ndarray]]:
     """
-    Fetch multiple channel timeseries from nds2 and store TimeSeriesDict,
+    Fetch multiple channel timeseries from nds2 and store in a TimeSeriesDict,
     or, if `array_lke` is True, a tuple of numpy arrays
     where the first element is an array of the channel data,
-    and the second element an array of corresponding times.
+    and the second element an array of corresponding times. If a channel name
+    is an interferometer name (e.g. `H1`), open data channels will be fetched.
+
     Thin wrapper around TimeSeriesDict.get
 
     Args:
         channels:
-            Channel names to fetch
-        t0:
+            List of channel names to fetch. If an interferometer
+            name is passed, (e.g. `H1`), open data channels will be fetched.
+        start:
             Start gpstime to read.
             If not passed will begin reading from earliest found time
-        tf:
+        end:
             Stop gpstime to read.
             If not passed will read until latest found time
-        nproc:
-            Number of concurrent processes to use with TimeSeriesDict.get
         array_like:
             Return in array like format. Otherwise, return gwpy.TimeSeriesDict
+        **kwargs:
+            key word arguments to pass to
+            TimeSeriesDict.get or TimeSeries.fetch_open_data
 
     Returns gwpy.TimeSeriesDict or Tuple of np.ndarrays
     """
-    ts_dict = TimeSeriesDict.get(
-        channels, start=t0, end=tf, nproc=nproc, verbose=True
+
+    open_data_channels = list(
+        filter(lambda x: x in OPEN_DATA_CHANNELS, channels)
     )
+    channels = list(filter(lambda x: x not in OPEN_DATA_CHANNELS, channels))
+
+    # fetch data from nds2
+    ts_dict = TimeSeriesDict()
+    if channels:
+        ts_dict = TimeSeriesDict.get(
+            channels,
+            start=start,
+            end=end,
+            verbose=verbose,
+            **kwargs,
+        )
+
+    # fetch open data channels and combine
+    if open_data_channels:
+        open_data_ts_dict = _fetch_open_data(
+            open_data_channels,
+            start=start,
+            end=end,
+            verbose=verbose,
+            **kwargs,
+        )
+        ts_dict.update(open_data_ts_dict)
+
+    print(ts_dict)
```

Comment:
Assuming this is just a missed debug statement

Reply:
Yup. happens too often